terraform/eks/agent.tf

```diff
@@ -0,0 +1,53 @@
+data "template_file" "cluster_role_file" {
+  template = file("./node-agent/cluster_role.tpl")
+  vars = {
+    NAMESPACE = "aoc-ns-${module.common.testing_id}"
+  }
+}
+
+data "template_file" "cluster_role_binding_file" {
+  template = file("./node-agent/cluster_role_binding.tpl")
+  vars = {
+    NAMESPACE       = "aoc-ns-${module.common.testing_id}",
+    SERVICE_ACCOUNT = kubernetes_service_account.aoc-agent-role.metadata[0].name
+  }
+}
+
+data "template_file" "config_map_file" {
+  template = file("./node-agent/config_map.tpl")
+  vars = {
+    NAMESPACE = kubernetes_namespace.aoc_ns.metadata[0].name
+  }
+}
+
+data "template_file" "daemonset_file" {
+  template = file("./node-agent/daemonset.tpl")
+  vars = {
+    NAMESPACE       = kubernetes_namespace.aoc_ns.metadata[0].name
+    SERVICE_ACCOUNT = kubernetes_service_account.aoc-agent-role.metadata[0].name
```

Comment:
So the service account is shared for all eks test cases? Just wondering if there is any issues because different test cases require different permissions. If two test cases are running at the same time, will this service account be assigned with different permissions.

Reply:
I don't think so. `ServiceAccount` is namespaced level resource. Also `ClusterRole` and `ClusterRoleBinding` are all suffixed with testing id (although I do feel we can downgrade RBAC into namespaced level too). As long as the two test cases are not executed in a same container at the same time, we'll be good.